service/matching/taskListManager.go

```diff
@@ -166,7 +165,15 @@ func newTaskListManager(
 			metrics.MatchingTaskListMgrScope,
 		))
 	}
-
+	var taskListTypeTag metrics.Tag
+	if taskList.taskType == persistence.TaskListTypeActivity {
+		taskListTypeTag = metrics.TaskListTypeTag("activity")
+	} else {
+		taskListTypeTag = metrics.TaskListTypeTag("decision")
+	}
+	tlMgr.pollerHistory = newPollerHistory(func(newSize int) {
+		tlMgr.metricScope().Tagged(taskListTypeTag).UpdateGauge(metrics.PollerPerTaskListCounter, float64(newSize))
```

Comment:
probably create the tagged scope outside the inline func to avoid creating a new scope every time the updateFunc is called.

Looks like we can update the `metricScope()` func to return a scope with taskListType tag and let other metrics benefit from the change? We don't do this to avoid breaking existing dashboards/alerts?